app/src/main/java/chat/rocket/android/chatroom/ui/ChatRoomAdapter.kt

```diff
@@ -55,22 +54,28 @@ class ChatRoomAdapter(private val serverUrl: String) : RecyclerView.Adapter<Chat
         notifyItemChanged(index)
     }
 
-    override fun getItemId(position: Int): Long {
-        return dataSet[position].id.hashCode().toLong()
-    }
-
-    class ViewHolder(itemView: View, val serverUrl: String) : RecyclerView.ViewHolder(itemView) {
+    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
 
-        fun bind(message: MessageViewModel) = with(itemView) {
-            bindUserAvatar(message, image_avatar, image_unknown_avatar)
-            text_user_name.text = message.sender
+        fun bind(message: MessageViewModel, nextMessage: MessageViewModel?) = with(itemView) {
+            image_avatar.setImageURI(message.avatarUri)
+            text_sender.text = message.senderName
             text_message_time.text = message.time
-            text_content.text = message.content
 
-            bindAttachment(message, message_attachment, image_attachment, audio_video_attachment,
-                    file_name)
+            if (nextMessage != null) {
```

Comment:
I think we should group messages on the ViewModel/ViewModel Mapper and keep the Adapter logic free. What do you thing @leonardoaramaki